html/controls/plt/plt-input.js

```diff
@@ -600,4 +623,113 @@ function preparePenaltyEditor() {
 		}
 	}
 }
+
+var annotationEditor;
+
+function openAnnotationEditor(teamId, skaterId) {
+	var prefix = 'ScoreBoard.Team('+teamId+').Skater('+skaterId+').';
+	var skaterNumber = WS.state[prefix + 'Number'];
+	var position = WS.state[prefix + 'Position'].slice(2);
+	var annotationPath = ').TeamJam('+teamId+').Fielding(' + position + ').Annotation';
+	if (isTrue(WS.state['ScoreBoard.InJam'])) {
+		annotationPath = 'ScoreBoard.Period(' + WS.state['ScoreBoard.CurrentPeriodNumber'] +
+		').Jam(' + WS.state['ScoreBoard.Period('+WS.state['ScoreBoard.CurrentPeriodNumber']+').CurrentJamNumber']
+		+ annotationPath;
+	} else {
+		annotationPath = 'ScoreBoard.Jam(' + WS.state['ScoreBoard.UpcomingJamNumber'] + annotationPath;
+	}
+	annotationEditor.data('skaterNumber', skaterNumber);
+	annotationEditor.data('position', position);
+	annotationEditor.find('#subDropdown').val(skaterId);
+	annotationEditor.find('#annotation').val(WS.state[annotationPath]);
+	annotationEditor.find('.Box').toggleClass('Hide', !isTrue(WS.state[prefix + 'PenaltyBox']));
+	annotationEditor.dialog('open');
+}
+
+function prepareAnnotationEditor(teamId) {
+	$(initialize);
+	
+	var subDropdown = $('<select>').attr('id', 'subDropdown');
+	var annotationField = $('<input type="text">').attr('size', '35').attr('id', 'annotation');
+	var jamPrefix;
+	
+	function initialize() {
+		var table = $('<table>').appendTo($('#AnnotationEditor'));
+		var row = $('<tr>').addClass('Box').appendTo(table);
+		$('<td>').append(subDropdown)
+				.append($('<button>').text('Substitute').button().click(function() {
+					var prefix = jamPrefix + annotationEditor.data('position') + ').';
+					WS.Set(prefix + 'Skater', subDropdown.val());
+					WS.Set(prefix + 'Annotation', 'Substitute for #' + annotationEditor.data('skaterNumber'));
+					annotationEditor.dialog('close');
+				})).appendTo(row);
+		$('<td>').append($('<button>').text('Self Report').button().click(function() { leaveBox('Self Report');})).appendTo(row);
+		$('<td>').append($('<button>').text('Penalty Revoked').button().click(function() { leaveBox('Penalty Revoked');})).appendTo(row);
```

Comment:
The revoked doesn't remove the penalty, so it still shows as waiting to serve.

Self report also removes the LT entry, which is incorrect.

Reply:
It's not possible to always correctly remove the penalty: It could be that the trip was for two penalties of which only one was removed. Also if PT and LT are staffed separately, this would allow the LT to meddle with the PT sheet, which I don't think is a good idea.

Self report simply ends the box trip at the point you press it. The only case where a box trip would vanish is if a skater enters and leaves in the same interjam as that means that it never started.

Reply:
> It's not possible to always correctly remove the penalty: It could be that the trip was for two penalties of which only one was removed.

In either case it should remove them from the Box.

Reply:
> In either case it should remove them from the Box.

It does.

Reply:
Hmm, can't reproduce now.